### PR TITLE
log object string correctly

### LINKED ISSUE #1584: Unexpected behavior when logging object with key `name`
### What version of Bun is running?

0.3.0

### What platform is your computer?

Linux 5.15.74.2-microsoft-standard-WSL2 #1 SMP Wed Nov 2 19:50:29 UTC 2022 x86_64 x86_64

### What steps can reproduce the bug?

```ts
console.log({ name: "bun" });
```

### How often does it reproduce? Is there a required condition?

No condition required, it's always the same output

### What is the expected behavior?

```
{name: 'bun'}
```

### What do you see instead?

```
bun {
  name: "bun"
}
```

The value of `name` in the object is printed infront of the object itself. The key has to be `name` and the value can be any `string`.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by alexlamsl**
use `Symbol.toStringTag` of the object or that of its `.prototype`

fixes #1584

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -701,12 +702,12 @@ USOCKETS_SRC_DIR = $(BUN_DEPS_DIR)/uws/uSockets/src/
...
-	cd $(USOCKETS_DIR) && $(CXX_WITH_CCACHE) -fno-builtin-malloc -fno-builtin-free -fno-builtin-realloc $(EMIT_LLVM_FOR_RELEASE) $(MACOS_MIN_FLAG)  -fPIC $(CXXFLAGS) $(UWS_CXX_FLAGS) -save-temps -I$(BUN_DEPS_DIR)/uws/uSockets/src $(UWS_LDFLAGS) -g $(DEFAULT_LINKER_FLAGS) $(PLATFORM_LINKER_FLAGS) $(OPTIMIZATION_LEVEL) -g -c $(wildcard $(USOCKETS_SRC_DIR)/*.cpp) $(wildcard $(USOCKETS_SRC_DIR)/**/*.cpp)
+	cd $(USOCKETS_DIR) && $(CC_WITH_CCACHE) -fno-builtin-malloc -fno-builtin-free -fno-builtin-realloc $(EMIT_LLVM_FOR_RELEASE)  $(MACOS_MIN_FLAG) -fPIC $(CFLAGS) $(UWS_CC_FLAGS) -save-temps -I$(BUN_DEPS_DIR)/uws/uSockets/src $(UWS_LDFLAGS) -g $(DEFAULT_LINKER_FLAGS) $(PLATFORM_LINKER_FLAGS) $(WRAP_SYMBOLS_ON_LINUX) $(OPTIMIZATION_LEVEL) -g -c $(wildcard $(USOCKETS_SRC_DIR)/*.c) $(wildcard $(USOCKETS_SRC_DIR)/**/*.c)
```
Comment (Jarred-Sumner) [inline Makefile]: won't need to do WRAP_SYMBOLS_ON_LINUX here, only need to do it in `bun-link-lld-debug` and `bun-link-lld-release`

Comment (alexlamsl) [reply Makefile]: fixed & tested in https://github.com/oven-sh/bun/pull/1674/commits/c3da1a4c1685616d78fe0076743742598decc960

**Comment by alexlamsl** [comment]
Please hold on &minus; found a bug.

**Comment by Jarred-Sumner** [comment]
I think using Symbol.toStringTag is a better approach. Classes are supposed to set that field automatically when defined.

File: test/bun.js/console/console-log.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { file, spawn } from "bun";
+import { expect, it } from "bun:test";
+import { bunExe } from "bunExe";
+
+it("should log to console correctly", async () => {
+  const { stdout, stderr, exited } = spawn({
```
Comment (Jarred-Sumner) [inline test/bun.js/console/console-log.test.ts:6]: if you just want to test console.log

We can use `Bun.inspect` for this

`Bun.inspect` runs the same code as console.log but outputs to a string

File: src/bun.js/bindings/exports.zig [hunk 3]
```diff
@@ -1704,6 +1703,13 @@ pub const ZigConsoleClient = struct {
                     value: JSValue,
                     is_symbol: bool,
                 ) callconv(.C) void {
+Output.prettyErrorln("KEY: {s} {d} {d}\nVALUE: {d} {d}", .{
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/exports.zig]: don't forget to not commit this

File: src/bun.js/bindings/bindings.cpp [hunk 4]
```diff
@@ -3075,9 +3074,6 @@ void JSC__JSValue__getNameProperty(JSC__JSValue JSValue0, JSC__JSGlobalObject* a
     }
 
     JSC::JSValue name = obj->getDirect(vm, vm.propertyNames->toStringTagSymbol);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.cpp]: we should probably switch this to getIfPropertyExists

Comment (alexlamsl) [reply src/bun.js/bindings/bindings.cpp]: Done in https://github.com/oven-sh/bun/pull/1674/commits/76f64875695528d6a5c83ebb88465cdca14e02b0

File: src/bun.js/bindings/exports.zig (hunk 3 above)
Comment (alexlamsl) [reply src/bun.js/bindings/exports.zig]: Purged in https://github.com/oven-sh/bun/pull/1674/commits/76f64875695528d6a5c83ebb88465cdca14e02b0

**Pull request merged**